R/barchart.R

```diff
@@ -393,17 +397,59 @@ Bar <- function(x,
         # Plotly text marker positions are not spaced properly when placed to
         # the left of the bar (i.e. negative values or reversed axis).
         # Adjusted by controlling the size of the marker
-        if (any(data.label.show))
+        if (any(data.label.show[,i]))
         {
-            p <- addDataLabelAnnotations(p, type = "Bar", legend.text[i],
+            # Add attribute for PPT exporting
+            ind.show <- which(data.label.show[,i])
+            chart.labels$SeriesLabels[[i]] <- list(
+                Font = setFontForPPT(data.label.font[[i]]), ShowValue = TRUE)
+
+            # Initialise custom points if annotations are used
+            pt.segs <- NULL
+            ind.show <- which(data.label.show[,i])
+            tmp.suffix <- if (percentFromD3(data.label.format)) sub("%", "", data.label.suffix[,i])
+                          else                                               data.label.suffix[,i]
+            if (!is.null(annotation.list) || length(ind.show) < nrow(chart.matrix) ||
+                any(nzchar(data.label.prefix[,i])) || any(nzchar(data.label.suffix[,i])))
+            {
+                chart.labels$SeriesLabels[[i]]$ShowValue <- FALSE
+                pt.segs <- lapply((1:nrow(chart.matrix)),
+                    function(ii) return(list(Index = ii-1, Segments = c(
```

Comment:
`return` statement not required. Can remove the `return` wrapper and get the same result.

Reply:
Removed `return` in a26f3eb